packages/server/src/api/controllers/automation.ts

```diff
@@ -159,6 +160,7 @@ export async function trigger(ctx: UserCtx) {
         automation,
         {
           fields: ctx.request.body.fields,
+          user: ctx.user as User,
```

Comment:
This is quite tragic isn't it. I had a look at why you needed to do this and `UserCtx` has a `ContextUser` (lol) as the `user` prop, and it's just an `Omit<User, "roles">`. Should `externalTrigger()` be updated to accept a `ContextUser` instead of `User`?

Reply:
Then we have to update the `User` all the way day into `AutomationDataEvent` to `ContextUser`. Which I guess makes sense, because it is a **_user that comes from context_**. But slightly disgusting all the same lol.